src/book_tests.rs

```diff
@@ -303,12 +224,37 @@ pub async fn test_deep_buy() {
     assert_eq!(ask_length, 3);
 }
 
+#[tokio::test]
```

Comment:
Would be worth testing the branch where the last order on the book is the users own order too

Reply:
Consult 4bd574e.